### PR TITLE
initial pass: users endpoint in oc_erchef

### PR DESCRIPTION
**Pull request by marcparadise**
- includes users endpoint support in erchef
- support in underyling components for oc_chef_wm requests
  that do not have a valid org, without stubbing out org.
- make oc_erchef aware of global containers in couch
- nginx routing for endpoints

/ping @opscode/server-team

Note that I'm leaning towards maintaining this as a feature branch
pending completion of other works in progress.

### CODE REVIEW

File: priv/pgsql_statements.config [hunk 1]
```diff
@@ -25,22 +25,21 @@
...
-      hashed_password, salt, hash_type, last_updated_by, created_at,
-      updated_at, external_authentication_uid,
+      pubkey_version, hashed_password, salt, hash_type, last_updated_by,
+      created_at, updated_at, external_authentication_uid,
       recovery_authentication_enabled, admin) VALUES
       ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, $14)">>}.
 
 {find_user_by_username,
-  <<"SELECT id, authz_id, username, email, public_key,
+  <<"SELECT id, authz_id, username, email, public_key, pubkey_version
```
Comment (seth) [inline priv/pgsql_statements.config:42]: For what it's worth, I think we don't need the pubkey_version field. Keeping it around might be the easiest thing for now, but turns out that you can look at the key and determine what you need to know.

**Comment by sdelano** [comment]
LGTM. The only decision here is whether or not we merge this or keep in in a feature branch. Once merged, we won't be able to ship chef_db in open source erchef.

File: priv/pgsql_statements.config (hunk 1 above)
Comment (marcparadise) [inline priv/pgsql_statements.config:42]: I did find that interesting - we go through this process to populate the field in the first place...

Comment (sdelano) [inline priv/pgsql_statements.config:42]: In order to keep back-compat, is it worth using the key itself to determine this information?

Comment (marcparadise) [inline priv/pgsql_statements.config:42]: With the additional of external_authentication to the insert/update fields in chef_user, I think we're still going to have backward-compat issues.

**Pull request merged**